WEB-INTEGRATION.md

```diff
@@ -0,0 +1,742 @@
+might therefore make sense to instead define an equivalent of that abstract
+operation in WebIDL, that handles this.
+
+Note that for properties of events and observer entries, implementations may
+allocate the actual `AsyncContext.Snapshot` instance lazily on first access, if
+they just store the internal pointer to the underlying snapshot when the event
+is created.
+
+## Using AsyncContext from web specs
+
+There are use cases in the web platform that would benefit from using
+AsyncContext variables built into the platform, since there are often relevant
+pieces of contextual information which would be impractical to pass explicitly
+as parameters. Some of these use cases are:
+
+- **Task attribution**. The soft navigations API
+  [\[SOFT-NAVIGATIONS\]](https://wicg.github.io/soft-navigations/) needs to be
+  able to track which tasks in the event loop are caused by other tasks, in
+  order to measure the time between the user interaction that caused the soft
+  navigation, and the end of the navigation. Currently this is handled by
+  modifying a number of event loop-related algorithms from the HTML spec, but
+  basing it on AsyncContext might be easier. It seems like this would also be
+  useful [to identify scripts that enqueued long
+  tasks](https://github.com/w3c/longtasks/issues/89), or to [build dependency
+  trees for the loading of
+  resources](https://github.com/w3c/resource-timing/issues/263). See
+  https://github.com/WICG/soft-navigations/issues/44.
+
+- **`scheduler.yield` priority and signal**. In order to provide a more
+  ergonomic API, if
+  [`scheduler.yield()`](https://wicg.github.io/scheduling-apis/#dom-scheduler-yield)
+  is called inside a task enqueued by
+  [`scheduler.postTask()`](https://wicg.github.io/scheduling-apis/#dom-scheduler-posttask)
+  [\[SCHEDULING-APIS\]](https://wicg.github.io/scheduling-apis/), its `priority`
+  and `signal` arguments will be “inherited” from the call to `postTask`. This
+  inheritance should propagate across awaits. See
+  https://github.com/WICG/scheduling-apis/issues/94.
+
+- **Future possibility: ambient `AbortSignal`**. This would allow using an
+  `AbortSignal` without needing to pass it down across the call stack until the
+  leaf async operations. See
+  https://gist.github.com/littledan/47b4fe9cf9196abdcd53abee940e92df
+
+- **Possible refactoring: backup incumbent realm**. The HTML spec infrastructure
+  for the [incumbent realm](https://html.spec.whatwg.org/multipage/webappapis.html#concept-incumbent-everything)
+  uses a stack of backup incumbent realms synchronized with the JS execution
+  stack, and explicitly propagates the incumbent realm through `await`s using JS
+  host hooks. This might be refactored to build on top of AsyncContext, which
+  might help fix some long-standing disagreements between certain browsers and
+  the spec.
+
+Each of these use cases would need to define `AsyncContext.Variable` instances
```

Comment:
Is this the right factoring? Or would it be better for there to be an abstract "async context variable record" or something that specs could use, without creating JS objects and dealing with all the associated fiddliness?

Reply:
ISTR we discussed factoring it like you're suggesting.